Add in some documentation around Live Templates. (#361)
TheBevyFlock/bevy_new_2d: docs/tooling.md

> If that is an issue for you, you can set it to `false` or remove the setting entirely.
> </details>

> ## RustRover/JetBrains Live Templates

Reviewer: ```suggestion
## RustRover Live Templates
```
Iirc other JetBrains IDEs stopped supporting Rust after RustRover was released.

Author: It's a bit more complex. RustRover is their main Rust IDE, but they also have a Rust Plugin that CLion, IntelliJ Idea, and some others support which gives Rust support to those. At least last I checked.

Reviewer: Ah ok I see. The Rust plugin for other JetBrains IDEs still exists but it's paid now.

> If you're using [RustRover](https://www.jetbrains.com/rust/), you might want to use Live Templates, which are a form of autocomplete template,
> which might make things easier. The documentation on how Live Settings work in RustRover can be found (here)[https://www.jetbrains.com/help/rust/using-live-templates.html]

Reviewer: ```suggestion
If you're using [RustRover](https://www.jetbrains.com/rust/), you may want to set up [Live Templates](https://www.jetbrains.com/help/rust/using-live-templates.html) to provide autocomplete for common boilerplate code.
```

Unfortunately, it is not really possible at this time to share Live Templates on a per-project basis, as they are global, however you can still set them up yourself.

> If you wish to have a quick guide for porting the existing snippets from vscode, then you can easily port the live snippets over by doing the following:

Reviewer: ```suggestion
Here's a quick guide for porting this template's [VS Code snippets](../.vscode/bevy.code-snippets) to Live Templates:
```

> - Replace any instances of $0 in the template with $END$
> - Replace any instances of $1 in the template with $NAME$ or something similar.

Reviewer: ```suggestion
- Replace any instances of `$0` in the template with `$END$`.
- Replace any instances of `$1` in the template with `$NAME$` or something similar.
```

- For the `plugin` template, you might want to set the applicability to rust modules.
- For the other templates, you might want to set the applicability to rust modules, statements, and expressions.

> to make it easier to enable/disable if you use RustRover for non-bevy applications you can also create a template group called Bevy and put the live templates under the group.

Reviewer: ```suggestion
To make it easier to enable or disable these live templates for different projects, you can put them in a template group called `Bevy`.
```
Optional alternative wording.
